youtube_dl/extractor/bbc.py

```diff
@@ -1074,6 +1086,85 @@ def _real_extract(self, url):
+        playlist = try_get(simorgh_data, lambda x: x['pageData']['promo']['media']['playlist']) or []
+        if playlist:
+            media = simorgh_data['pageData']['promo']
+            if media['media'].get('format') == 'video':
+                media.update(media['media'])
+                formats = []
+                keys = {'url', 'format', 'format_id', 'language', 'quality', 'tbr', 'resolution'}
+                for format in playlist:
+                    if not (format.get('url') and format.get('format')):
+                        continue
+                    bitrate = format.pop('bitrate')
+                    if bitrate:
+                        bitrate = re.sub(r'000\s*$', 'kbps', bitrate)
+                        format['tbr'] = parse_bitrate(bitrate)
+                    format['language'] = media.get('language')
+                    # format id: penultimate item from the url split on _ and .
+                    (fmt,) = re.split('[_.]', format['url'])[-2:][:1]
+                    format['format_id'] = '%s_%s' % (format['format'], fmt)
+                    if not format.get('resolution'):
+                        format['resolution'] = fmt
+                    format['quality'] = -1
+                    formats.append(dict((k, format[k]) for k in keys))
+                self._sort_formats(formats)
+                return {
+                    'id': media.get('id'),
+                    'title': (dict_get(media.get('headlines'),
+                                       ('shortHeadline', 'headline'))
+                              or playlist_title),
+                    'description': media.get('summary') or playlist_description,
+                    'formats': formats,
+                    'subtitles': None,
+                    'thumbnail': try_get(media, lambda x: x['image']['href']),
+                    'timestamp': int_or_none(media.get('timestamp'), scale=1000)
+                }
+
+        # general case: media nested in content object
+        # test: https://www.bbc.co.uk/scotland/articles/cm49v4x1r9lo
+        if simorgh_data:
+
+            def extract_media_from_simorgh(model):
+                if not isinstance(model, dict):
+                    return
+                for block in model.get('blocks') or {}:
+                    if block.get('type') == 'aresMediaMetadata':
+                        vpid = try_get(block, lambda x: x['model']['versions'][0]['versionId'])
+                        if vpid:
+                            formats, subtitles = self._download_media_selector(vpid)
+                            self._sort_formats(formats)
+                            model = block['model']
+                            version = model['versions'][0]
+                            thumbnail = model.get('imageUrl')
+                            return {
+                                'id': vpid,
+                                'title': model.get('title') or 'unnamed clip',
```

Comment:
Is this really possible?

Reply:
Or throw on `model['title']`. But these pages may have a lot of clips and I don't know whether the BBC will set the title every time.